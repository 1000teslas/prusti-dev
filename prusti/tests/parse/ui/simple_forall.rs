/// Tests that parser handles spans correctly.

extern crate prusti_contracts;


#[requires="forall x: i32, y: usize :: {x + 2, x + 3; x + 4} x > 0 ==> x + 2 > 2"]
pub fn test1a(x: i32) {}

#[requires="forall x: 32, y: usize :: {} x > 0 ==> x > -1"]
pub fn test1b(x: i32) {}

#[requires="forall"]
pub fn test1c(x: i32) {}

#[requires="forall x: i32, y: usize :: {x + 2, x + 3; x + 4} x > 0 ==> x + 2 > 2
    ==> true"]
pub fn test1d(x: i32) {}


fn main() {}
