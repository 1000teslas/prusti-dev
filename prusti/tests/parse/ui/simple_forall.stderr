error: failed to parse forall bounded variable list
 --> $DIR/simple_forall.rs:7:19
  |
7 | #[requires="forall x: 32, y: usize :: {} x > 0 ==> x > -1"]
  |                   ^^^^^^^^^^^^^^^^

error: failed to parse forall expression
  --> $DIR/simple_forall.rs:10:12
   |
10 | #[requires="forall"]
   |            ^^^^^^^^

error: forall can have only one implication
  --> $DIR/simple_forall.rs:13:12
   |
13 |   #[requires="forall x: i32, y: usize :: {x + 2, x + 3; x + 4} x > 0 ==> x + 2 > 2
   |  ____________^
14 | |     ==> true"]
   | |_____________^

error: aborting due to 3 previous errors

